File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/Fragment1.java
package com.example.lifecyclesandbox;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.fragment.app.Fragment;

public class Fragment1 extends Fragment implements View.OnClickListener {

    View view;

    public Fragment1() {
        super(R.layout.fragment_1);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        view = inflater.inflate(R.layout.fragment_1, container, false);
        Button loginButton = (Button) view.findViewById(R.id.login_button);
        loginButton.setOnClickListener(this);
        return view;
    }

    @Override
    public void onClick(View v) {
        // Display Fragment2
        new Fragment2().show(getChildFragmentManager(), Fragment2.TAG);
    }
}

File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/login/Fragment1.kt
package com.example.lifecyclesandbox.login

import com.example.lifecyclesandbox.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.fragment.app.Fragment
import com.example.lifecyclesandbox.login.Fragment2

class Fragment1 : Fragment(R.layout.fragment_1), View.OnClickListener {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_1, container, false)
        val loginButton = view.findViewById<View>(R.id.login_button) as Button
        loginButton.setOnClickListener(this)
        return view
    }

    override fun onClick(v: View) {
        // Display Fragment2
        Fragment2().show(childFragmentManager, Fragment2.TAG)
    }
}